kapitan/targets.py

```diff
@@ -324,6 +340,46 @@ def compile_target(target_obj, search_paths, compile_path, ref_controller, **kwa
     logger.info("Compiled %s (%.2fs)", target_name, time.time() - start)
 
 
+def get_dependencies_per_target(target_objs):
+    """returns a mapping from target name to a list of dependency objects"""
+    mapping = {}
+    for target_obj in target_objs:
+        try:
+            dependencies = target_obj["dependencies"]
+            target_name = target_obj["vars"]["target"]
+            mapping[target_name] = []
+            for item in dependencies:
+                dependency_type = item["type"]
+                source_uri = item["source"]
+                output_path = item["output_path"]
+                if dependency_type == "git":
+                    del item["output_path"]
```

Comment:
Is this deleted because for git dependencies you still cache? 
If so, might be good to add a small comment here for it. :+1:

Reply:
@uberspot it is deleted because we have a parameter called output_path in the constructor, while also having output_path as the key in the item dict. When I pass in **item as kwargs then it would throw error for duplicate arguments for output_path. I could just make subdir and ref parameters explicit in the constructor (defaulting to none) as well. Let me know what you think.

Reply:
This is ok, just add a code comment to make it clearer for future reviewers too. :)

Reply:
Yep, please add comment! It was also not obvious to me :)